src/main/java/com/google/devtools/build/lib/analysis/actions/FileWriteAction.java

```diff
@@ -114,20 +116,23 @@ public static FileWriteAction createEmptyWithInputs(
    * @param fileContents the contents to be written to the file
    * @param makeExecutable whether the output file is made executable
    * @param allowCompression whether (transparent) compression is enabled
+   * @param mnemonic an optional custom mnemonic for the action, or null to use the default
    */
   public static FileWriteAction create(
       ActionOwner owner,
       Artifact output,
       CharSequence fileContents,
       boolean makeExecutable,
-      Compression allowCompression) {
+      Compression allowCompression,
+      @Nullable String mnemonic) {
```

Comment:
I think that instead of adding @Nullable everywhere, it would be better to create a new public static method with non-nullable mnemonic, and keep the other methods untouched.